### PR TITLE
feat: Use data types defined in the Rust Wren Engine to do type mapping

### PR DESCRIPTION
**Pull request by onlyjackfrost**
Use data types defined in the Rust Wren Engine to do type mapping

### CODE REVIEW

File: ibis-server/app/model/metadata/trino.py [hunk 1]
```diff
@@ -97,38 +98,40 @@ def _get_schema_name(self):
...
-            "bool": WrenEngineColumnType.BOOLEAN,
-            "boolean": WrenEngineColumnType.BOOLEAN,
+            "bool": RustWrenEngineColumnType.BOOL,
+            "boolean": RustWrenEngineColumnType.BOOL,
             # Decimal
-            "float": WrenEngineColumnType.FLOAT8,
-            "double": WrenEngineColumnType.DOUBLE,
-            "decimal": WrenEngineColumnType.DECIMAL,
-            "numeric": WrenEngineColumnType.NUMERIC,
+            "float": RustWrenEngineColumnType.FLOAT8,
```
Comment (goldmedal) [inline ibis-server/app/model/metadata/trino.py:125]:
```suggestion
            "float": RustWrenEngineColumnType.FLOAT4,
```

**Review by goldmedal — APPROVED** [review]
Thanks @onlyjackfrost overall LGTM. Only one suggestion.

File: ibis-server/app/model/metadata/trino.py (hunk 1 above)
Comment (onlyjackfrost) [reply ibis-server/app/model/metadata/trino.py]: [8ffae51](https://github.com/Canner/wren-engine/pull/913/commits/8ffae51295cfded3302a8091d1c4c0c3c9e4275e)

**Review by grieve54706 — APPROVED** [review]
Nice to have tests.

**Pull request merged**